### PR TITLE
Migrate the function to controller.

### PR DESCRIPTION
**Pull request by teweitsai**
* Migrate the functions to **Controller** class in **ts_m2com**.

### CODE REVIEW

File: python/lsst/ts/m2/csc.py [hunk 1]
```diff
@@ -156,18 +157,16 @@ async def set_mount_elevation_callback(self, data: salobj.BaseMsgType) -> None:
...
-                comp_name="MTMount",
-            )
+            await self.controller_cell.set_external_elevation_angle(data.actualPosition)
 
     async def set_mount_elevation_in_position_callback(
         self, data: salobj.BaseMsgType
     ) -> None:
         """Callback function to notify the mount elevation in position.
 
+        Note. This function will be removed after the CSC communicates with
```
Comment (tribeiro) [inline python/lsst/ts/m2/csc.py:167]: Consider using a [deprecation warning](https://numpydoc.readthedocs.io/en/latest/format.html#deprecation-warning) instead.

Basically change this to:


```
.. deprecated:: 0.8.0
          This function will be removed after the CSC communicates with the cRIO directly.
```

at the end of the docstring.

File: python/lsst/ts/m2/csc.py [hunk 2]
```diff
@@ -684,27 +689,32 @@ def _assert_enabled_csc_and_controller(
         self.controller_cell.assert_controller_state(message_name, allowed_curr_states)
         self.assert_enabled()
 
-    async def _write_command_to_server(
-        self, message_name: str, timeout: float, message_details: dict | None = None
+    async def _execute_command_to_server(
```
Comment (tribeiro) [inline python/lsst/ts/m2/csc.py:692]: I think I liked the previous name better, this new name reads a bit weird. Maybe remove "_to_server" and just call it `_execute_command`?

File: python/lsst/ts/m2/csc.py [hunk 3]
```diff
@@ -684,27 +689,32 @@ def _assert_enabled_csc_and_controller(
...
+
         try:
-            await self.controller_cell.write_command_to_server(
-                message_name,
-                message_details=message_details,
-                timeout=timeout,
-            )
+            await command(*args, timeout=timeout, **kwargs)  # type: ignore[operator]
 
         except OSError:
```
Comment (tribeiro) [inline python/lsst/ts/m2/csc.py:719]: I know this is not part of the PR but maybe document in the docstring that the method captures `OSError` and sends the CSC to fault. Also, consider logging the exception before sending the CSC to fault, e.g.;

```
except OSError:
    self.log.exeception("Connection error executing command.")
    await self.controller_cell.close()
    await self.fault(...)
```

**Review by tribeiro — APPROVED** [review]
Looks good! I left a few inline comments for you to consider.

File: python/lsst/ts/m2/csc.py (hunk 1 above)
Comment (teweitsai) [reply python/lsst/ts/m2/csc.py]: Updated in the commit: [71466d9](https://github.com/lsst-ts/ts_m2/pull/45/commits/71466d9802e0a0c6b2ae58e37845e0cd3430fc94). Thanks!

File: python/lsst/ts/m2/csc.py (hunk 2 above)
Comment (teweitsai) [reply python/lsst/ts/m2/csc.py]: Renamed in the commit: [71466d9](https://github.com/lsst-ts/ts_m2/pull/45/commits/71466d9802e0a0c6b2ae58e37845e0cd3430fc94). Thanks!

File: python/lsst/ts/m2/csc.py (hunk 3 above)
Comment (teweitsai) [reply python/lsst/ts/m2/csc.py:722]: Updated in the commit: [71466d9](https://github.com/lsst-ts/ts_m2/pull/45/commits/71466d9802e0a0c6b2ae58e37845e0cd3430fc94). Thanks!

**Pull request merged**